x/evm/types/params.go

```diff
@@ -171,6 +171,10 @@ func validateNetworks(network interface{}) error {
 		if n.Name == "" {
 			return sdkerrors.Wrap(types.ErrInvalidGenesis, "network name cannot be an empty string")
 		}
+
+		if !n.Id.IsPositive() {
```

Comment:
is 0 not allowed?

Reply:
Not as far as I can tell. Existing chains have a non-zero value, the spec is unclear about it, so I thought it's better to be on the safer side. Could always revert this check in an upgrade if really needed.
@sammy1991106 thoughts?

Reply:
chain id being 0 is valid but creates a replayable transaction.